scheds/c/scx_central.c

```diff
@@ -49,7 +49,17 @@ int main(int argc, char **argv)
 	struct bpf_link *link;
 	__u64 seq = 0, ecode;
 	__s32 opt;
-	cpu_set_t *cpuset;
+	__u32 max_cpu_ids, nr_cpu_ids = 0, i = 0;
+	cpu_set_t *cpuset, curcpuset;
+
+	max_cpu_ids = libbpf_num_possible_cpus();
+	CPU_ZERO(&curcpuset);
+	SCX_BUG_ON(sched_getaffinity(0, sizeof(curcpuset), &curcpuset),
+		   "Failed to get current CPU topology");
+	for (i = 0; i < max_cpu_ids; i ++) {
+		if (CPU_ISSET(i, &curcpuset))
+			nr_cpu_ids ++;
```

Comment:
`nr_cpu_ids` is maximum possible CPU + 1. The above is counting the number of CPUs in the current task's affinity, which is a different thing. You'd have to go by `/sys/devices/system/cpu/possible`.